music21/romanText/tsvConverter.py

```diff
@@ -0,0 +1,670 @@
+            raise ValueError("Direction must be 'm21-DCML' or 'DCML-m21'.")
+
+        if 'vii' in preString.lower():
+            position = preString.lower().index('vii')
+            prevChar = preString[position-1] # the previous character,  # / b.
+            if prevChar == search:
+                postString = preString[:position-1]+preString[position:]
+            else:
+                postString = preString[:position]+insert+preString[position:]
+        else:
+            postString = preString
+
+    return postString
+
+def getLocalKey(local_key, global_key, convert=False):
+    '''
+    Re-casts comparative local key (e.g. 'V of G major') in its own terms ('D').
+    >>> romanText.tsvConverter.getLocalKey('V', 'G')
+    'D'
+    >>> romanText.tsvConverter.getLocalKey('ii', 'C')
+    'd'
+
+    By default, assumes an m21 input, and operates as such:
+    >>> romanText.tsvConverter.getLocalKey('vii', 'a')
+    'g#'
+
+    Set convert=True to convert from DCML to m21 formats. Hence;
+    >>> romanText.tsvConverter.getLocalKey('vii', 'a', convert=True)
+    'g'
+    '''
+
+    if convert==True:
+        local_key = characterSwaps(local_key, minor=is_minor(global_key[0]), direction='DCML-m21')
+
+    asRoman = roman.RomanNumeral(local_key, global_key)
+    rt = asRoman.root().name
+    if asRoman.isMajorTriad():
+        newKey = rt.upper()
+    elif asRoman.isMinorTriad():
+        newKey = rt.lower()
+
+    return newKey
+
+def vLocalKey(rn, local_key):
+    '''
+    Separates comparative roman numeral for tonicisiations like 'V/vi' into the component parts of
+    - a roman numeral (V) and
+    - a (very) local key (vi)
+    and expresses that very local key in relation to the local key also called (DCML column 11).
+
+    >>> romanText.tsvConverter.getLocalKey('vi', 'C')
+    'a'
+    >>> romanText.tsvConverter.vLocalKey('V/vi', 'C')
+    'a'
+    '''
+
+    if '/' not in rn:
+        very_local_as_key = local_key
+    else:
+        position = rn.index('/')
+        very_local_as_roman = rn[position+1:]
+        very_local_as_key = getLocalKey(very_local_as_roman, local_key)
+
+    return very_local_as_key
+
+# ------------------------------------------------------------------------------
+
+class Test(unittest.TestCase):
+
+    def runTest(self):
+        pass
+
+    def testTsvHandler(self):
+
+        name = 'tsvEgOp18No1Mov3.tsv'
+        path = common.getSourceFilePath() / 'romanText' / name
+
+        importedTsv = TsvHandler(path)
+        # Based on the DCML file, with minor error correction (removing '#viio' of major keys);
+        # Also includes a '@none' / rest test case;
+        # No time signature changes though - tested separately.
+
+        # Raw
+        self.assertEqual(importedTsv.tsvData[0][0], '.F.I')
+        self.assertEqual(importedTsv.tsvData[35][0], '#viio6/ii')
+
+        # Chords
+        importedTsv.tsvToChords()
+        testTabChord1 = importedTsv.chordList[0] # Also tests makeTabChord()
```

Comment:
how would a user know that they should call `tsvToChords()` and then look at `handler.chordList`?